compiler/lib/code.ml

```diff
@@ -308,7 +308,12 @@ type prim_arg =
   | Pc of constant
 
 type expr =
-  | Apply of Var.t * Var.t list * bool
+  | Apply of
+      { f : Var.t
+      ; args : Var.t list
+      ; exact : bool
+      ; tailcall : bool
```

Comment:
It seems that tailcall is only used in  `Generate.translate_expr`. I wonder if it would be better to reconstruct the information on the fly by passing additional information in `Generate.translate_instrs`.
I like the use of inline record but maybe we don't need to have tailcall in it.

Reply:
The set of tail calls is now stored separately. Does this look better to you? I should probably revert the type of `Code.Apply` as well to minimize the changes in the PR.

Reply:
It looks better to me but I still wonder if it would be better to compute tailcalls in generate.ml directly. That way, we don't have to worry about other passes messing up with tailcalls after applying `Effects.t`.
 `Generate.Share.get` already performs a full traversal and could additionally track tailcalls. Is there something I'm missing that forces us to compute tailcalls in Effects ?

Reply:
> I should probably revert the type of Code.Apply as well to minimize the changes in the PR.

I find the code clearer with such change. I'm happy to keep it.

Reply:
For now, we can compute tailcalls in `generate.ml` directly, indeed.

But if we CPS-transform the program only partially, we will need to keep track somehow which parts have been transformed. But hopefully the inlining phase after `Effects.f` will then no longer be necessary.

Reply:
maybe the name `tailcall` is not precise enough / too generic..